tools/bin/go_core_tests_integration

```diff
@@ -4,8 +4,42 @@ set +e
 
 SCRIPT_PATH=$(dirname "$0"); SCRIPT_PATH=$(eval "cd \"$SCRIPT_PATH\" && pwd")
 OUTPUT_FILE=${OUTPUT_FILE:-"./output.txt"}
+JUNIT_FILE=${JUNIT_FILE:-"./junit.xml"}
+GO_TEST_TIMEOUT=${GO_TEST_TIMEOUT:-"10m"}
 EXTRA_FLAGS=""
 
+# Install gotestsum as test runner
+echo "Installing gotestsum..."
+go install gotest.tools/gotestsum@v1.12.3
+PATH=$PATH:$(go env GOPATH)/bin
+export PATH
+
+# Setup test flags based on event type
+echo "Running go_core_tests_integration for event: $GITHUB_EVENT_NAME"
+GO_TEST_FLAGS="-timeout=${GO_TEST_TIMEOUT}"
```

Comment:
For building args/flags, it's generally safer to make it an array.
```suggestion
GO_TEST_FLAGS=( -timeout "${GO_TEST_TIMEOUT}" )
```

And then just add to it:

```
GO_TEST_FLAGS+=( -count=1 )
```

And then eventually use it like:

```
-- --tags integration "${GO_TEST_FLAGS[@]}" ...
```